core/types/block.go

```diff
@@ -982,3 +1105,66 @@ type HashAndNumber struct {
 	Hash   common.Hash
 	Number uint64
 }
+
+type BestPhKey struct {
```

Comment:
Does this all belong in `block.go`? It seems unrelated to block structure. If anything, this seems specific to how slice.go is managing pending headers. Or maybe its a module of its own.

Reply:
will put this in its own module